open-api/rest-catalog-open-api.yaml

```
@@ -2747,6 +2747,54 @@ components:
         uuid:
           type: string
 
+    AzureCredentials:
+      type: object
+      properties:
+        account-name:
+          type: string
+        account-key:
+          type: string
+        token:
+          type: string
+
+    AwsCredentials:
+      type: object
+      required:
+        - access-key-id
+        - secret-access-key
+      properties:
+        access-key-id:
+          type: string
+        secret-access-key:
+          type: string
+        token:
+          type: string
+        expires-at-ms:
+          type: integer
+          format: int64
+
+    GcsCredentials:
+      type: object
+      required:
+        - token
+        - expires-at
+      properties:
+        token:
+          type: string
+        expires-at-ms:
+          type: integer
+          format: int64
+
+    Credentials:
+      type: object
+      properties:
+        aws:
+          $ref: '#/components/schemas/AwsCredentials'
+        azure:
+          $ref: '#/components/schemas/AzureCredentials'
+        gcs:
+          $ref: '#/components/schemas/GcsCredentials'
```

We might also add a type to it, this is a bit more flexible to extend I believe (you could add a custom credential in there that will be picked up by a custom FileIO):
```suggestion
    Credentials:
      discriminator:
        propertyName: type
        mapping:
          aws: '#/components/schemas/AwsCredentials'
          azure: '#/components/schemas/AzureCredentials'
          gcs: '#/components/schemas/GcsCredentials'
      type: object
      required:
        - type
      properties:
        type:
          type: string
```

---

good point, that makes sense